code/game/machinery/landline.dm

```diff
@@ -0,0 +1,393 @@
+	attached_to = A
+	linked_phone = new phone_type (src)
+	linked_phone.linked_landline = src
+	phone = linked_phone
+	
+	phone_overlay = image(icon = overlay_icon, icon_state = overlay_iconstate)
+	attached_to.overlays.Add(phone_overlay) //TODO make this less shit
+
+/obj/landline/proc/delete_cord()
+	if(!linked_phone)
+		return
+	for(var/obj/C in linked_cord)
+		qdel(C)
+	linked_cord = list()
+
+/obj/landline/proc/make_cord()
+	if(!linked_phone)
+		return
+	if(get_dist(src, linked_phone) > tether_length-1)
+		linked_phone.visible_message("\the [src] cord stretches dangerously")
+		if(get_dist(src, linked_phone) > tether_length)
+			linked_phone.visible_message("<span class='warning'>the cord snaps!</span>")
+			linked_phone.linked_landline = null
+			linked_phone = null		
+			return
+			
+	var/obj/cable1
+	var/obj/cable2
+	var/turf/T = attached_to.loc
+	var/first_offset = FALSE
+	if(src.attached_to.pixel_x < -15)
+		first_offset = WEST
+	if(src.attached_to.pixel_x > 15)
+		first_offset = EAST
+	if(src.attached_to.pixel_y < -15)
+		first_offset = SOUTH
+	if(src.attached_to.pixel_y > 15)
+		first_offset = NORTH
+	if(first_offset)
+		cable1 = new /obj/effect/phone_cord (T)
+		cable1.dir = first_offset
+		linked_cord += cable1
+	var/list/getstepto = get_steps_to(src.attached_to, linked_phone.loc)
+	for(var/D in getstepto)
+		cable2 = new /obj/effect/phone_cord (T)
+		cable2.dir = D
+		T = get_step(T,D)
+		cable1 = new /obj/effect/phone_cord (T)
+		cable1.dir = turn(D,180)
+		linked_cord += cable1
+		linked_cord += cable2
+		
+/obj/landline/proc/shake_phone_overlay(amplitude = 2)
+	if(!phone)
+		return
+	spawn(0)
+		var/pixel_x_diff = rand(-amplitude, amplitude) * PIXEL_MULTIPLIER
+		var/pixel_y_diff = rand(-amplitude, amplitude) * PIXEL_MULTIPLIER
+		animate(attached_to, pixel_x = attached_to.pixel_x + pixel_x_diff, pixel_y = attached_to.pixel_y + pixel_y_diff , time = 1, loop = 10,easing = BOUNCE_EASING)
+		animate(pixel_x = attached_to.pixel_x - pixel_x_diff, pixel_y = attached_to.pixel_y - pixel_y_diff , time = 1, loop = 10,easing = BOUNCE_EASING)
+	
+/obj/landline/proc/has_power()
+	var/obj/machinery/requests_console/RC = attached_to
+	if(istype(RC, /obj/machinery/requests_console) && RC.stat)
+		return FALSE
+	return TRUE //redphones stay powered regardless
+	
+/obj/landline/proc/end_call_loop()
+	spawn(0)
+		while(linked_phone && !phone && has_power())
+			playsound(source=linked_phone, soundin=linked_phone.end_call_sound, vol=100, vary=FALSE, channel=0)
+			sleep(1 SECONDS)
+	
+/obj/landline/proc/ring_loop()
+	if(calling && calling.phone)
+		calling.ringing = TRUE
+	spawn(0)
+		while(calling && calling.calling == src && calling.ringing)
+			if(!has_power())
+				calling.ringing = FALSE
+			calling.ring()
+			if(linked_phone && !phone)
+				playsound(source=linked_phone, soundin=linked_phone.dial_sound, vol=100, vary=FALSE, channel=0)
+				//this sound is barely perceptible if you're not right next to the phone

```

Comment:
People already ignore RC's as-is: you should make this loud and obnoxious as I am sure it will rarely be used

Reply:
This is a **dialing** sound, not a ringing sound. Only you, the caller, are supposed to hear this.

On that note, ringing IS moderately loud and can be heard from across the rooms (watch the vidrel).